appinventor/components/src/com/google/appinventor/components/runtime/util/TextViewUtil.java

```diff
@@ -175,7 +176,12 @@ public static String getText(TextView textview) {
    * @param text  new text to be shown
    */
   public static void setTextHTML(TextView textview, String text) {
-    textview.setText(Html.fromHtml(text));
+    if (Build.VERSION.SDK_INT >= 24 {
+      textView.setText(Html.fromHtml(text, Html.FROM_HTML_MODE_COMPACT));
```

Comment:
What is the rational of `FROM_HTML_MODE_COMPACT` here, versus `FROM_HTML_MODE_LEGACY` which the Android docs say is equivalent to the behavior prior to Nougat?

[reference](https://developer.android.com/reference/android/text/Html?hl=en#fromHtml(java.lang.String))

Reply:
Legacy: Separate block-level elements with blank lines (two newline characters) in between. This is the legacy behavior prior to N.

Compact: Separate block-level elements with line breaks (single newline character) in between. This inverts the Spanned to HTML string conversion done with the option TO_HTML_PARAGRAPH_LINES_INDIVIDUAL.

However, I only added these lines because it is said to use `fromHtml(String source, int flags`.

Reply:
Given that the HTML is rendered I think this doesn't matter in the end, other than a slight decrease in memory due to the use of single newlines rather than double newlines.